src/util/logger/mod.rs

```diff
@@ -447,37 +360,120 @@ fn test_log_format() {
         }
     }
 
-    // Make the log
-    let decorator = PlainSyncDecorator::new(TestWriter);
-    let drain = TikvFormat::new(decorator).fuse();
-    let logger = slog::Logger::root_typed(drain, slog_o!());
-    slog_error!(logger, "test");
-
-    // Check the logged value.
-    BUFFER.with(|buffer| {
-        let buffer = buffer.borrow_mut();
-        let output = from_utf8(&*buffer).unwrap();
-
-        // This functions roughly as an assert to make sure that the log level and file name is logged.
-        let mut split_iter = output.split(" ERRO mod.rs:");
-        // The pre-split portion will contain a timestamp which we can check by parsing and ensuring it is valid.
-        let datetime = split_iter.next().unwrap();
+    #[test]
+    fn test_log_format() {
+        use regex::Regex;
+        use std::time::Duration;
+        let decorator = PlainSyncDecorator::new(TestWriter);
+        let drain = TikvFormat::new(decorator).fuse();
+        let logger = slog::Logger::root_typed(drain, slog_o!());
+
+        // Empty message is not recommend. Just for test purpose.
+        slog_info!(logger, "");
+        slog_info!(logger, "Welcome");
+        slog_info!(logger, "Welcome TiKV");
+        slog_info!(logger, "欢迎");
+        slog_info!(logger, "欢迎 TiKV");
+
+        let duration = Duration::new(3, 0);
+        slog_info!(logger, "failed to fetch URL";
+                    "url"=> "http://example.com",
```

Comment:
pay attention to the format. Usually we should write:

```
"key" => ...
     ^  ^
     space
```

Reply:
Nice catch! `make format` not work here.🤪